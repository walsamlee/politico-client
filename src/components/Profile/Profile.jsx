/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';

const Profile = () => {
  let userImage = localStorage.getItem('picture');
  if (userImage === '') {
    userImage = 'https://res.cloudinary.com/walsam/image/upload/v1559156362/user-1633249__340.png';
  }

  return (
    <div className="user-info">
      <div id="user-info-main" className="user-info-main">
        <div className="user-profile-details">
          <div className="profile-pic">
            <img src={userImage} alt="Profile" />
          </div>
          <div id="profile-details" className="profile-details">
            <h3>{localStorage.getItem('name')}</h3>
            <p>{localStorage.getItem('email')}</p>
            <p>{localStorage.getItem('phone')}</p>
            <button type="button" className="btn btn-cart">Edit profile</button>
          </div>
        </div>
        <div className="office-interest">
          <h3>Run for Office</h3>
          <form method="POST" action="">
            <div className="party-selector">
              <label htmlFor="party">Party: </label>
              <select name="party">
                <option value="0">-------------Select-----------</option>
                <option value="1">Peoples Democratic Party</option>
                <option value="2">Actions Congress of Nigeria</option>
                <option value="3">Labour Party</option>
                <option value="4">All Democratic Congress</option>
              </select>
            </div>
            <div className="office-selector">
              <label htmlFor="office">Office: </label>
              <select name="office">
                <option value="0">-------------Select-----------</option>
                <option value="President">President</option>
                <option value="Senator">Senator</option>
                <option value="Governor">Governor</option>
                <option value="Chairman">LGA Chairman</option>
              </select>
            </div>
            <div>
              <input type="button" className="btn btn-cart" value="Submit" />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default Profile;
