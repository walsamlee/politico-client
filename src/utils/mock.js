const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
