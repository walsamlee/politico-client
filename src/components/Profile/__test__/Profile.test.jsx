import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import Profile from '../Profile';

const { profileData } = mock;

describe('<Profile />', () => {
  it('should render Profile component', () => {
    const wrapper = shallow(
      <Profile party={profileData} />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
