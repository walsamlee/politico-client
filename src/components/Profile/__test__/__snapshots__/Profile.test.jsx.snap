// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render Profile component 1`] = `ShallowWrapper {}`;
